File: reactive.infra.form.engine/src/main/java/org/infra/reactive/form/engine/form/engine/model/dto/response/button/CoreButtonAssignElementDTO.java
package org.infra.reactive.form.engine.form.engine.model.dto.response.button;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.infra.reactive.form.engine.form.engine.model.dto.response.css.CoreCssDTO;
import org.infra.reactive.form.engine.form.engine.model.dto.response.element.CoreAllElementDTO;
import org.infra.reactive.form.engine.form.engine.model.dto.response.element.CoreAllElementExtraAttributeValueDTO;
import org.infra.reactive.form.engine.form.engine.model.dto.response.layout.CoreLayoutDataAssignElementDTO;

import java.io.Serializable;
import java.util.Map;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CoreButtonAssignElementDTO implements Serializable {
    private long id;
    private String name;
    private String translate;
    private CoreButtonDTO coreButtonDTO;
    private CoreCssDTO coreCssDTO;
    private CoreAllElementDTO coreAllElementDTO;
    private Long recordId;
    private Integer buttonIndex;
    private Map<Long, CoreLayoutDataAssignElementDTO> coreLayoutDataAssignElementDTOMap;
    private Map<Long, CoreAllElementExtraAttributeValueDTO> coreAllElementExtraAttributeValueDTOMap;
    private CoreAllElementDTO coreAllElementDTOModule;
    private Long recordIdModule;
}

File: reactive.infra.web/src/main/resources/static/Resources/JS/ERPFrameWork/Communicate/Models/Response/Button/CoreButtonAssignElementDTO.js
import {BaseModel} from "../../../../../UIFrameWork/Shared/Common/BaseModel.js";
import ConvertUtil from "../../../Common/ConvertUtil.js";
import CoreAllElementDTO from "../Element/CoreAllElementDTO.js";
import CoreButtonDTO from "./CoreButtonDTO.js";
import CoreCssDTO from "../Css/CoreCssDTO.js";
import CoreLayoutDataAssignElementDTO from "../Layout/CoreLayoutDataAssignElementDTO.js";
import CoreAllElementExtraAttributeValueDTO from "../Element/CoreAllElementExtraAttributeValueDTO.js";

export default class CoreButtonAssignElementDTO extends BaseModel {
    constructor() {
        super();
    }

    getCoreButtonDTO() {
        if (this.coreButtonDTO && !(this.coreButtonDTO.constructor.prototype instanceof BaseModel)) {
            this.coreButtonDTO = ConvertUtil.ConvertGeneral(CoreButtonDTO, this.coreButtonDTO);
        }
        return this.coreButtonDTO;
    }

    getCoreCssDTO() {
        if (this.coreCssDTO && !(this.coreCssDTO.constructor.prototype instanceof BaseModel)) {
            this.coreCssDTO = ConvertUtil.ConvertGeneral(CoreCssDTO, this.coreCssDTO);
        }
        return this.coreCssDTO;
    }

    getCoreAllElementDTO() {
        if (this.coreAllElementDTO && !(this.coreAllElementDTO.constructor instanceof BaseModel)) {
            this.coreAllElementDTO = ConvertUtil.ConvertGeneral(CoreAllElementDTO, this.coreAllElementDTO);
        }
        return this.coreAllElementDTO;
    }

    getRecordId() {
        return this.recordId;
    }

    getButtonIndex() {
        return this.buttonIndex;
    }

    setCoreLayoutDataAssignElementDTOMap(coreLayoutDataAssignElementDTOMap) {
        this.coreLayoutDataAssignElementDTOMap = coreLayoutDataAssignElementDTOMap;
    }

    getCoreLayoutDataAssignElementDTOMap() {
        if (this.coreLayoutDataAssignElementDTOMap && !(this.coreLayoutDataAssignElementDTOMap instanceof Map)) {
            this.coreLayoutDataAssignElementDTOMap = ConvertUtil.ConvertGeneralWithMap(CoreLayoutDataAssignElementDTO, this.coreLayoutDataAssignElementDTOMap);
        }
        return this.coreLayoutDataAssignElementDTOMap;
    }

    getCoreAllElementExtraAttributeValueDTOMap() {
        if (this.coreAllElementExtraAttributeValueDTOMap && !(this.coreAllElementExtraAttributeValueDTOMap instanceof Map)) {
            this.coreAllElementExtraAttributeValueDTOMap = ConvertUtil.ConvertGeneralWithMap(CoreAllElementExtraAttributeValueDTO, this.coreAllElementExtraAttributeValueDTOMap);
        }
    }

    getCoreAllElementDTOModule() {
        if (this.coreAllElementDTOModule && !(this.coreAllElementDTOModule.constructor instanceof BaseModel)) {
            this.coreAllElementDTOModule = ConvertUtil.ConvertGeneral(CoreAllElementDTO, this.coreAllElementDTOModule);
        }
        return this.coreAllElementDTOModule;
    }

    getRecordIdModule() {
        return this.recordIdModule;
    }
}